src/components/transactions/FlowCommons/TxModalDetails.tsx

```diff
@@ -98,42 +115,58 @@ export const DetailsNumberLineWithSub = ({
-        <Box sx={{ display: 'flex', alignItems: 'center' }}>
-          {value && (
-            <>
-              <FormattedNumber value={value} variant="secondary14" color={color} />
-              {!hideSymbolSuffix && (
-                <Typography ml={1} variant="secondary14">
-                  {symbol}
-                </Typography>
-              )}
-              <SvgIcon color="primary" sx={{ fontSize: '14px', mx: 1 }}>
-                <ArrowNarrowRightIcon />
-              </SvgIcon>
-            </>
-          )}
-          <FormattedNumber value={futureValue} variant="secondary14" color={color} />
-          {!hideSymbolSuffix && (
-            <Typography ml={1} variant="secondary14">
-              {symbol}
-            </Typography>
-          )}
-        </Box>
-        <Box sx={{ display: 'flex', alignItems: 'center' }}>
-          {valueUSD && (
-            <>
-              <FormattedNumber value={valueUSD} variant="helperText" compact symbol="USD" />
-              <SvgIcon color="primary" sx={{ fontSize: '14px', mx: 1 }}>
-                <ArrowNarrowRightIcon />
-              </SvgIcon>
-            </>
-          )}
-          <FormattedNumber value={futureValueUSD} variant="helperText" compact symbol="USD" />
-        </Box>
+        {loading ? (
+          <Skeleton variant="rectangular" height={20} width={100} sx={{ borderRadius: '4px' }} />
+        ) : (
+          <Box sx={{ display: 'flex', alignItems: 'center' }}>
+            {value && (
+              <>
+                <FormattedNumber value={value} variant="secondary14" color={color} />
+                {!hideSymbolSuffix && (
+                  <Typography ml={1} variant="secondary14">
+                    {symbol}
+                  </Typography>
+                )}
+                <SvgIcon color="primary" sx={{ fontSize: '14px', mx: 1 }}>
+                  <ArrowNarrowRightIcon />
+                </SvgIcon>
+              </>
+            )}
+            {tokenIcon && <TokenIcon symbol={tokenIcon} sx={{ mr: 1, fontSize: '14px' }} />}
+            <FormattedNumber value={futureValue} variant="secondary14" color={color} />
+            {!hideSymbolSuffix && (
+              <Typography ml={1} variant="secondary14">
+                {symbol}
+              </Typography>
+            )}
+          </Box>
+        )}
+        {loading ? (
+          <Skeleton
+            variant="rectangular"
+            height={15}
+            width={80}
+            sx={{ borderRadius: '4px', marginTop: '4px' }}
+          />
+        ) : (
+          <Box sx={{ display: 'flex', alignItems: 'center' }}>
+            {valueUSD && (
+              <>
+                <FormattedNumber value={valueUSD} variant="helperText" compact symbol="USD" />
+                <SvgIcon color="primary" sx={{ fontSize: '14px', mx: 1 }}>
+                  <ArrowNarrowRightIcon />
+                </SvgIcon>
```

Comment:
This `<SvgIcon>` is used several times throughout this file. Consider creating a const for this and reusing it the other 6 times.